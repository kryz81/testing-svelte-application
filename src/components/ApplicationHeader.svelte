<script>
  export let appVersion = '';
</script>

<h2 class="application-header">My Expense Tracker {appVersion}</h2>
