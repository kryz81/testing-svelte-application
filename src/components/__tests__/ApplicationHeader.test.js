import ApplicationHeader from "../ApplicationHeader";

const createComponent = (Component, props) => {
  const target = document.body.appendChild(document.createElement("div"));
  return {
    renderedElement: target,
    component: new Component({ target, props: { ...props } })
  };
};

beforeEach(() => {
  document.body.innerHTML = "";
});

it("renders without error", async () => {
  const result = () => createComponent(ApplicationHeader, { appVersion: 'v1.0' });
  expect(result).not.toThrowError();
});

it("correctly renders the component", () => {
  const { renderedElement } = createComponent(ApplicationHeader, { appVersion: 'v1.0' });
  expect(renderedElement).toMatchSnapshot();
});

it("correctly renders the app header", () => {
  const { renderedElement } = createComponent(ApplicationHeader, { appVersion: 'v1.0' });
  const result = renderedElement.getElementsByTagName("h1");
  expect(result[0].innerHTML).toEqual("My Expense Tracker v1.0");
});
