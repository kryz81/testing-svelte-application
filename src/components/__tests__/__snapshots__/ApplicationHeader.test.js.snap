// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders the component 1`] = `
<div>
  <h1>
    My Expense Tracker 
    v1.0
  </h1>
</div>
`;
